### DISCUSSION TITLE
How Github fails the fundamentals of usability

### DISCUSSION THREAD

**Discussion by Dawoodoz** [community/community#57882] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

* Due to cramming in more components than what fits inside the browser, the web interface keeps jerking around, so that when you try to press something, it jumps and your work is lost. It is rare to get the link that you aimed for.
* Github accepts careful scrolling with a stylus pen as a hard mouse press, without any additional pressure added.
* A common way to undo a click in progress is to pull away the cursor before releasing, yet Github accepts pulling away as explicitly confirming the click.
* Before saving something, it is common to ask for confirmation, but Git does not care about user consent.
* Now you have a confusing piece of history taking up valuable space next to your comments, about selecting something that you did not even click to begin with.
* While you can undo an attribute change in a pull request, doing so doubles the problem by putting in a big notification about having removed it seconds later. Further hiding the useful information in the generic clutter of notifications that nobody even care about.
* Now that your comment field is cluttered with lots of history that should not have been there to begin with, because it provides no useful information to the visitors, Github won't even let you delete the pull request, even though you have already merged the exact same code in another pull request and the PR is not even a part of the git history.

If any of these bugs would be fixed, Github would be tolerable as a free product, but slowly adding more bugs while letting users get used to more and more crazy workarounds to get the simplest task done seems to be the core market strategy.

**Comment by hamidrezamd1380** [comment]
https://realit.ir/
